### PR TITLE
Added Nqueens Problem solution

### PR DESCRIPTION
**Pull request by shreeshjha**
I've provided a solution for Nqueens, along with some notes clarifying what the problem is and what results we may expect when we run the programme.

### CODE REVIEW

**Review by siriak — APPROVED** [review]
Please run `cargo fmt --all` to fix linting errors

File: README.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ These are for demonstration purposes only.
 ## [General](./src/general)
 
 - [x] [Convex Hull: Graham Scan](./src/general/convex_hull.rs)
-- [ ] N-Queensp
+- [x] [N-Queensp](./src/general/nqueens.rs)
```
Comment (imp2002) [inline README.md:63]:
```suggestion
- [x] [N-Queens](./src/general/nqueens.rs)
```
the  alphabet `p`? Maybe it's a typo and it also appeared in README.

File: src/general/mod.rs [hunk 2]
```diff
@@ -1,8 +1,11 @@
 mod convex_hull;
 mod hanoi;
 mod kmeans;
+mod nqueens;
 
 pub use self::convex_hull::convex_hull_graham;
 pub use self::hanoi::hanoi;
 pub use self::kmeans::f32::kmeans as kmeans_f32;
 pub use self::kmeans::f64::kmeans as kmeans_f64;
+pub use self::nqueens::nqueens;
```
Comment (imp2002) [inline src/general/mod.rs:10]: no fn nqueens in nqueens.rs

File: src/general/nqueens.rs [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	    if det {
+		    board[r][i] = true;
+		    t(&mut board, r + 1, &mut cnt);
+		    board[r][i] = false;
+	    }
+    }
+
+}
+
+fn main() {
```
Comment (imp2002) [inline src/general/nqueens.rs:42]: use test👀

**Review by imp2002 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**